client/src/controllers/BulkController.test.js

```diff
@@ -0,0 +1,58 @@
+import { Application } from '@hotwired/stimulus';
+import { BulkController } from './BulkController';
+
+describe('BulkController', () => {
+  beforeEach(() => {
+    document.body.innerHTML = `
+    <div data-controller="w-bulk">
+      <input type="checkbox" data-w-bulk-target="all" data-action="w-bulk#toggleAll">
+      <div>
+        <input type="checkbox" data-w-bulk-target="item" disabled data-action="w-bulk#toggle">
+        <input type="checkbox" data-w-bulk-target="item" data-action="w-bulk#toggle">
+        <input type="checkbox" data-w-bulk-target="item" data-action="w-bulk#toggle">
+      </div>
+      <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="false">Clear all</button>
+      <button data-action="w-bulk#toggleAll" data-w-bulk-force-param="true">Select all</button>
+    </div>
+    `;
+    const application = Application.start();
+    application.register('w-bulk', BulkController);
+  });
+
+  it('executes the correct action when the select all checkbox is clicked', () => {
+    const allCheckbox = document.querySelector('[data-w-bulk-target="all"]');
+    const toggleAllActionSpy = jest.spyOn(
+      BulkController.prototype,
+      'toggleAll',
+    );
+
+    allCheckbox.click();
+
+    expect(toggleAllActionSpy).toHaveBeenCalled();
+  });
+
+  it('executes the correct action when an item checkbox is clicked', () => {
+    const itemCheckbox = document.querySelector(
+      '[data-w-bulk-target="item"]:not([disabled])',
+    );
+    const toggleActionSpy = jest.spyOn(BulkController.prototype, 'toggle');
+
+    itemCheckbox.click();
+
+    expect(toggleActionSpy).toHaveBeenCalled();
+  });
+
+  it('executes the correct action when the Clear all button is clicked', () => {
+    const clearAllButton = document.querySelector(
+      '[data-action="w-bulk#toggleAll"][data-w-bulk-force-param="false"]',
+    );
+    const toggleAllActionSpy = jest.spyOn(
+      BulkController.prototype,
+      'toggleAll',
+    );
+
+    clearAllButton.click();
+
+    expect(toggleAllActionSpy).toHaveBeenCalled();
+  });
+});
```

Comment:
```suggestion
  it('executes the correct action when the Set all button is clicked', () => {
    const allCheckbox = document.getElementById('select-all');
    const setAllButton = document.getElementById('set');

    expect(allCheckbox.checked).toBe(false);

    expect(
      document
        .getElementById('checkboxes')
        .querySelectorAll(':checked:not(:disabled)').length,
    ).toEqual(0);

    setAllButton.click();

    expect(
      document
        .getElementById('checkboxes')
        .querySelectorAll(':checked:not(:disabled)').length,
    ).toEqual(2);

    expect(allCheckbox.checked).toBe(true);
  });
});
```

Finally, one more test for the set all button, you added it to the test HTML, let's ensure we test it works as expected.

It's actually really important to check things when they are set to `true`, as often things will work ok when the falsey state is happening but not when the truthey state is happening.